### ISSUE TITLE
Training on Custom dataset

### ISSUE THREAD

**Issue by sree3333** [dbolya/yolact#282]
I have a dataset of 4000 images on which I have to train with pretrained weights (let's say yolact_im700_54_800000.pth), I have made changes in backbone.py "strict=True" as told in issue https://github.com/dbolya/yolact/issues/27 . But still, nothing improved after training. It got worse

could someone help me how to train on a new dataset with pretrained weights or how to do transfer learning?

**Comment by dbolya** [comment]
You might want to lower the learning rate before fine tuning, since you'd be overwriting the pretrained results otherwise.

Admittedly, the model isn't set up very well right now for fine tuning since you'd likely want to only train certain layers instead of training the whole model, so proper fine tuning support is on the TODO list I guess.

**Comment by sree3333** [comment]
Hey @dbolya 
I have tried decreasing the learning rate as you told me but there is no improvement in segmentation and also result of the bounding box got bad.

You can see in the result attached below
![output_image1](https://user-images.githubusercontent.com/17790112/74404244-ed5e8900-4e4f-11ea-962b-8357ec82fa60.png)

**Comment by dbolya** [comment]
You might want to try something like in #334.

Also make sure you're not training too long. With that small of a dataset, 40k training iterations is probably enough.

**Comment by sree3333** [comment]
Thank you so much for your advice @dbolya ..I will get back to you once I m done with the training!

**Comment by sree3333** [comment]
Hey @dbolya , I have tried the method you told me.. I think it will get better by fine-tuning
 the model with adjusting learning rate n decay.
But In some results, I am getting double segmentation for the same object as shown below!

Could you tell me why this could be happening
![front49](https://user-images.githubusercontent.com/17790112/74918241-ce7f6a00-53ee-11ea-9741-5748dfec3877.png)

**Comment by sree3333** [comment]
Hey @dbolya , I have tried the method you told me.. I think it will get better by fine-tuning
the model with adjusting learning rate n decay.
But In some results, I am getting double segmentation for the same object as shown below!

Could you tell me why this could be happening
![front49](https://user-images.githubusercontent.com/17790112/74918494-2fa73d80-53ef-11ea-8162-0f8341b35cf3.png)
![front71](https://user-images.githubusercontent.com/17790112/74918611-5796a100-53ef-11ea-8ab0-87a2a92e58e6.png)

**Comment by Auth0rM0rgan** [comment]
change `cross_class_nms` from `False` to `True` (`--cross_class_nms=True`)

**Comment by AndreiBaraian** [comment]
Hey, I am also trying to do transfer learning on a custom dataset. My questions are:
1. In what range should the lr be? Currently I am using 1e-4
2. I start from a pre-trained yolac model (resnet101), but I get 
![image](https://user-images.githubusercontent.com/24776877/75018334-5f118500-5497-11ea-98ce-8062f8a13cde.png)

even after training for quite some time.
Do you have any intuition on what could be going wrong? And thanks a lot for this amazing repo @dbolya !

**Comment by sree3333** [comment]
> You might want to try something like in #334.
> 
> Also make sure you're not training too long. With that small of a dataset, 40k training iterations is probably enough.

I have done as you told me with learning rate 
![Screenshot from 2020-02-24 09-06-39](https://user-images.githubusercontent.com/17790112/75139932-e703c300-5713-11ea-8da5-44bde6ad86b3.png)
at 1e -5 and a dataset of 2000 with only a single class but still, I m getting very low mAP for the mask.
![imageedit_1_9614102188](https://user-images.githubusercontent.com/17790112/75140850-bb81d800-5715-11ea-895b-b4968054cc1f.png)
![imageedit_2_2360355228](https://user-images.githubusercontent.com/17790112/75140863-c3417c80-5715-11ea-985c-9e548da12ad2.png)
And also some wrong detections
![front115](https://user-images.githubusercontent.com/17790112/75141402-dd2f8f00-5716-11ea-8ebb-2f9d13b0c62d.png)
No detection in this picture

**Comment by dbolya** [comment]
@sree3333 Are you sure your annotations are correct? It's really weird how it almost perfectly segments out some random object and calls it a person, but not the actual person in the center of the image. 

And for
> I have done as you told me with learning rate

Did you add the `.detatch()` in #334?

**Comment by sree3333** [comment]
Hi @dbolya , thanks for your quick responses
Yes, I have added .detatch() in #334

Our annotations are good we have checked once again!

I also checked these images with pre-trained weights, it is giving the same result ..It is detecting some other random thing. 
Could you please why this could happen! 

Thanks a ton for helping me out.

**Comment by dbolya** [comment]
Sorry for the delay on this one. Are you still having this issue? If so, what does the output of the model look like on these images if you don't modify the original code, just use the off the shelf COCO models. The COCO models already should be able to detect people well, so there might be something wrong with your dataset definition.
